### PR TITLE
(FINERACT-241) Include "Add Note" to Deposit and Withdrawal screen

### PR DESCRIPTION
**Pull request by ippezshelby**
@nazeer1100126 , This is the final PR as guided by you. i tested it and it works fine, You can merge it

### CODE REVIEW

**Comment by nazeer1100126** [comment]
@Ippezrobert Don't we require to show this note in the UI when user is selecting these transactions?

**Comment by ippezshelby** [comment]
@nazeer1100126 , there is a PR i sent for the UI https://github.com/openMF/community-app/pull/2191. Hope this is what you meant.

**Comment by nazeer1100126** [comment]
@Ippezrobert in this screen https://demo.openmf.org/#/viewsavingtrxn/790/trxnId/9212

**Comment by ippezshelby** [comment]
@nazeer1100126 , Yes i get it. Indeed we need to show it in th above UI and i was thinking also in transaction tab https://demo.openmf.org/#/viewsavingaccount/790 as in this https://mifosforge.jira.com/secure/attachment/32487/Savings%20Trans%23%20View.png and https://mifosforge.jira.com/secure/attachment/32488/Loan%20Trans%23%20View.png

**Comment by nazeer1100126** [comment]
@Ippezrobert are you going to add this and send another PR?

**Comment by ippezshelby** [comment]
@nazeer1100126 , you can guide me of which changes or files are affected and i can add it as before and send a PR. Waiting for your guidance

**Comment by ippezshelby** [comment]
@nazeer1100126 , i have started working on this and almost done but i would like some assistance on the below mapping syntax in SavingsAccountTransactionsMapper class found in SavingsAccountReadPlatformServiceImpl.java

LEFT JOIN (SELECT mn.savings_account_id id, mn.note Note FROM m_note mn) notes ON notes.id=tr.savings_account_id

Just correct this and am done. Thanks

**Comment by nazeer1100126** [comment]
@Ippezrobert can we discuss on Skype ?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/DepositAccountReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -981,6 +983,12 @@ public SavingsAccountTransactionsMapper() {
             sqlBuilder.append("left join m_account_transfer_transaction totran on totran.to_savings_transaction_id = tr.id ");
             sqlBuilder.append("left join m_payment_detail pd on tr.payment_detail_id = pd.id ");
             sqlBuilder.append("left join m_payment_type pt on pd.payment_type_id = pt.id ");
+			sqlBuilder.append("left join m_appuser au on au.id=tr.appuser_id ");
+			sqlBuilder.append("left join m_note mn on mn.savings_account_id=sa.id ");
+			sqlBuilder.append("left join (");
+            sqlBuilder.append("select mn.savings_account_id id, mn.note Note");
+            sqlBuilder.append(" from m_note mn ");
+            sqlBuilder.append(") comments ON comments.id=tr.savings_account_id  ");
```
Comment (ippezshelby) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/DepositAccountReadPlatformServiceImpl.java:991]: @nazeer1100126 , we didn't have the skype call but however i tried to work things out. Then challenge i got is that the query is not joining the note with the transaction id correctly. Please have a look at it here before i make changes to send final PR. Thanks
What happens is that it repeats one note for all transactions not particular to a transaction

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountReadPlatformServiceImpl.java [hunk 2]
```diff
@@ -791,7 +793,13 @@ public SavingsAccountTransactionsMapper() {
...
             sqlBuilder.append("left join m_payment_detail pd on tr.payment_detail_id = pd.id ");
             sqlBuilder.append("left join m_payment_type pt on pd.payment_type_id = pt.id ");
-
+			sqlBuilder.append("left join m_appuser au on au.id=tr.appuser_id ");
+			sqlBuilder.append("left join m_note mn on mn.savings_account_id=sa.id ");
+			sqlBuilder.append("left join (");
+            sqlBuilder.append("select mn.savings_account_id id, mn.note Note");
+            sqlBuilder.append(" from m_note mn ");
+            sqlBuilder.append(") comments ON comments.id=tr.savings_account_id  ");
+
```
Comment (ippezshelby) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountReadPlatformServiceImpl.java:802]: @nazeer1100126 , we didn't have the skype call but however i tried to work things out. Then challenge i got is that the query is not joining the note with the transaction id correctly. Please have a look at it here before i make changes to send final PR. Thanks
What happens is that it repeats one note for all transactions not particular to a transaction

**Comment by ippezshelby** [comment]
@nazeer1100126 , Any comment on the above to help me with the PR?
Thanks

**Comment by nazeer1100126** [comment]
@Ippezrobert I believe we need to implement savings transaction note similar to loan transaction note.
1) Add a new column in m_note table for savings transaction id 'savings_transaction_id'
 @ManyToOne
    @JoinColumn(name = "savings_transaction_id", nullable = true)
    private SavingsAccountTransaction savingsTransaction;
2) Add a new note type in NoteType -> SAVINGS_TRANSACTION(800, "noteType.loan.transaction", "loanTransactions")
3) Create savings transaction note with newly created savings account transaction.
public static Note savingsTransactionNote(final SavingsAccount savingsAccount, final SavingsAccountTransaction savingsTransaction, final String note) {
        return new Note(savingsAccount, savingsTransaction, note);
}
4) Write the queries which are similar to loan transaction notes for savings transaction notes.

**Pull request closed without merging**